services/QuillLMS/app/workers/identify_stripe_invoices_without_subscriptions_worker.rb

```diff
@@ -29,14 +29,35 @@ def perform
   private def relevant_stripe_invoices
     stripe_invoices.filter do |invoice|
       RELEVANT_INVOICE_STATUSES.include?(invoice.status) &&
-        !Subscription.exists?(stripe_invoice_id: invoice.id)
+        !invoice_ids_with_subscriptions.include?(invoice.id) &&
+        !invoice_refunded?(invoice) &&
```

Comment:
What about switching the order of `non_zero_amount?` and `invoice_refunded?`.

The reason being that if `invoice_refunded?` fails it will short circuit the cycle and skip the API call to `Stripe::Charge.retrieve`